test/hotspot/jtreg/serviceability/jvmti/vthread/SelfSuspendDisablerTest/SelfSuspendDisablerTest.java

```diff
@@ -51,26 +57,57 @@ private static void sleep(long millis) {
         }
     }
 
+    private static void testJvmtiThreadState(Thread thread, int expectedState) throws RuntimeException {
+        String kindStr = thread.isVirtual()? "virtual " : "platform";
+        int state = getThreadState(thread);
+
+        System.out.printf("Expected %s thread state: %06X got: %06X\n",
+                          kindStr, expectedState, state);
+        if (state != expectedState) {
+            throw new RuntimeException("Test FAILED: Unexpected thread state");
+        }
+    }
+
     public static void main(String argv[]) throws Exception {
-        Thread t1 = Thread.ofPlatform().start(() -> {
+        Thread t1 = Thread.ofPlatform().factory().newThread(() -> {
+            testJvmtiThreadState(Thread.currentThread(), RUNNABLE);
             selfSuspend();
         });
-        Thread t2 = Thread.ofVirtual().start(() -> {
+        Thread t2 = Thread.ofVirtual().factory().newThread(() -> {
+            testJvmtiThreadState(Thread.currentThread(), RUNNABLE);
             while(!isSuspended(t1)) {
               Thread.yield();
             }
             Thread.yield(); // provoke unmount
+
+            testJvmtiThreadState(t1, SUSPENDED);
+
             resume(t1);
+
+            testJvmtiThreadState(t1, RUNNABLE);
+
             suspendAllVirtualThreads();
         });
 
+        testJvmtiThreadState(t1, NEW);
+        testJvmtiThreadState(t2, NEW);
+
+        t1.start();
+        t2.start();
+
         while(!isSuspended(t2)) {
             sleep(100);
         }
+
+        testJvmtiThreadState(t2, SUSPENDED);
```

Comment:
Not a useful check after the isSuspended(t2) call above, but no harm in it either.

Reply:
You are right in both cases. :)